core/src/main/java/org/infinispan/persistence/spi/CacheWriter.java

```diff
@@ -35,4 +38,28 @@
     * @throws PersistenceException in case of an error, e.g. communicating with the external storage
     */
    boolean delete(Object key);
+
+   /**
+    * Persist all provided entries to the store in a single batch update. If this is not supported by the
+    * underlying store, then entries are written to the store sequentially via {@link #write(MarshalledEntry)}
```

Comment:
I originally went with a List as I thought going forward it would provide us with more flexibility for future use-cases where we may need to write a batch containing duplicates. However, our present use cases (passivation, putAll, async writer) guarantee that there will be no duplicates. So I I'm happy with List or Collection depending on how strong you feel about it